README.md

```
@@ -39,203 +29,314 @@ import { transpileToPineJS } from '@opusaether/pine-transpiler';
 
-### MACD Indicator
+## Architecture
 
-```typescript
-const macdCode = `
-//@version=5
-indicator("MACD", overlay=false)
+The transpiler follows a classic compiler pipeline with four distinct phases:
 
-[macdLine, signalLine, histLine] = ta.macd(close, 12, 26, 9)
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)